### PR TITLE
Added a Pipeline steps feild 

### PR DESCRIPTION
**Pull request by ARSHIL1804**
- added pipeline-steps field for #2483

### CODE REVIEW

File: packages/twenty-front/src/generated-metadata/graphql.ts [hunk 1]
```diff
@@ -179,7 +179,8 @@ export enum FieldMetadataType {
   Relation = 'RELATION',
   Select = 'SELECT',
   Text = 'TEXT',
-  Uuid = 'UUID'
+  Uuid = 'UUID',
+  PipelineSteps = 'PIPELINE_STEPS'
```
Comment (charlesBochet) [inline packages/twenty-front/src/generated-metadata/graphql.ts:183]: PIPELINE_STEP should be singular as a fieldType. Could you replace pipelineSteps by pipelineStep everywhere in your code when talking about the field type?

File: packages/twenty-front/src/modules/object-record/field/meta-types/display/components/PipelineStepsDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Status } from '@/ui/display/status/components/Status';
+
+import { usePipelineStepsField } from '../../hooks/usePipelineStepsField';
+
+export const PipelineStepsDisplay = () => {
+  const { fieldValue } = usePipelineStepsField();
+
+  return <Status color={fieldValue.color} text={fieldValue.label} />;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/field/meta-types/display/components/PipelineStepsDisplay.tsx:8]: the fieldValue should be a string (we only want to store the value). The color and the label should be retrieved through metadata (see how Select is implemented)

File: packages/twenty-front/src/modules/object-record/field/meta-types/hooks/usePipelineStepsField.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  const { fieldName } = fieldDefinition.metadata;
+
+  const [fieldValue, setFieldValue] = useRecoilState<FieldPipelineStepsValue>(
+    entityFieldsFamilySelector({
+      entityId: entityId,
+      fieldName: fieldName,
+    }),
+  );
+  const fieldPipelineStepsValue = isFieldPipelineStepsValue(fieldValue)
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/field/meta-types/hooks/usePipelineStepsField.ts:32]: same everywhere! value should only be the value (see Select)

File: packages/twenty-front/src/modules/object-record/field/types/guards/isFieldPipelineStepsValue.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { z } from 'zod';
+
+import { themeColorSchema } from '@/ui/theme/utils/themeColorSchema';
+
+const pipelineStepsValueSchema = z.object({
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/field/types/guards/isFieldPipelineStepsValue.ts:5]: same here (see Select)

File: packages/twenty-front/src/modules/settings/data-model/components/SettingsObjectFieldPipelineStepsFormOptionRow.tsx [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+  align-items: center;
+  display: flex;
+  height: ${({ theme }) => theme.spacing(6)};
+  padding: ${({ theme }) => theme.spacing(1.5)} 0;
+`;
+
+const StyledColorSample = styled(ColorSample)`
+  cursor: pointer;
+  margin-left: 9px;
+  margin-right: 14px;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/settings/data-model/components/SettingsObjectFieldPipelineStepsFormOptionRow.tsx:45]: This margin-left 9px and margin-right 14px looks suspicious! Double check the figma to see what we really want and use theme.spacing()

File: packages/twenty-front/src/modules/settings/data-model/components/SettingsObjectFieldPipelineStepsFormOptionRow.tsx [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+`;
+
+const StyledOptionRowNumber = styled.div`
+  color: ${({ theme }) => theme.font.color.extraLight};
+  font-size: 13px;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/settings/data-model/components/SettingsObjectFieldPipelineStepsFormOptionRow.tsx:50]: use theme for all px values (theme.fontSize, theme.spacing()...) see other components :)

**Review by charlesBochet — COMMENTED** [review]
Nice work and thank you! I have left comments, I haven't tested the product behavior yet.
I can also see some duplicated code with Select that looks very similar on the Settings part. I feel we can share a bit more components between the two

**Comment by charlesBochet** [comment]
Also, we will need to add storybook tests on this section

File: packages/twenty-front/src/modules/object-record/field/meta-types/display/components/PipelineStepsDisplay.tsx (hunk 2 above)
Comment (ARSHIL1804) [reply packages/twenty-front/src/modules/object-record/field/meta-types/display/components/PipelineStepsDisplay.tsx:8]: > the fieldValue should be a string (we only want to store the value). The color and the label should be retrieved through metadata (see how Select is implemented)

Sorry for that as I haven't checked the latest update.

**Comment by ARSHIL1804** [comment]
@charlesBochet  all the suggested changes has been done.

**Comment by charlesBochet** [comment]
@ARSHIL1804 I'm really sorry, we had a lot of discussions on PipelineStep field types recently and we have decided to deprecate it as it is too close to a Select.

Your work is high quality and I apologize for the waste. If you still feel motivated, we could deprecate PipelineStep in the whole codebase as a follow up

**Comment by charlesBochet** [comment]
Sure, I'll create an issue shortly

**Comment by charlesBochet** [comment]
@ARSHIL1804 while writing the technical ticket, I actually figured out that we already have removed it. So we are all good!

**Pull request closed without merging**